### PR TITLE
Provision

### PR DESCRIPTION
**Pull request by abessiari**
Added docker file and aws provisionning.

See provision/PROVISION_README.md for local testing
and PROVISION_AWS_README.md for aws.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ADD ./ ./
+
+ARG uid=1000
+ARG gid=1000
+
+ENV MEM="2G"
+
+RUN echo '#!/bin/bash\n\
+set -x\n\
+java -server -Djetty.port=8899 -Xmx$MEM -Djetty.overrideWebXml=./conf/readonly_cors.xml -Dbigdata.propertyFile=./conf/blazegraph.properties -cp jars/blazegraph-jar.jar:jars/jetty-servlets.jar com.bigdata.rdf.sail.webapp.StandaloneNanoSparqlServer \
```
Comment (kltm) [inline docker/Dockerfile:21]: @abessiari Hm. Where are these files coming in from?

**Comment by kltm** [comment]
@abessiari I've started testing this and it's looking pretty good. One thing I noticed, when working with a larger query, was that I seemed to get timeouts after a few (5?) seconds:
`java.util.concurrent.ExecutionException: java.util.concurrent.ExecutionException: org.openrdf.query.QueryInterruptedException: java.lang.RuntimeException: java.util.concurrent.ExecutionException: com.bigdata.bop.engine.QueryTimeoutException: Query deadline is expired.`
We'll probably want to parameterize timeouts. I was looking to see how to do that, but couldn't figure out where the config files were (https://github.com/geneontology/go-graphstore/pull/19/files#r749868678).

**Comment by abessiari** [comment]
@kltm 

conf/readonly_cors.xml    there is timeout set to 30 seconds. 

<context-param>
    <param-name>queryTimeout</param-name>
    <param-value>30000</param-value>
 </context-param>

**Comment by kltm** [comment]
@abessiari Ah, thank you--I'm not sure how I missed that.

**Comment by kltm** [comment]
@abessiari With the testing that I'm doing and the timeouts I keep hitting, I think it would be useful if we could parameterize the `queryTimeout` like we do for the RAM. If that's tricky, I'll try and make up an SOP so we can do this at need.

**Comment by kltm** [comment]
@abessiari Similar to the reported issue with the other provisioning, I'm running into:
```
sjcarbon@moiraine:~/local/src/git/go-graphstore-abessiari/provision[provision]$:( ansible-playbook -e "stage_dir=$STAGE_DIR" -u ubuntu -i "$HOST," --private-key $PRIVATE_KEY build_image.yaml 

PLAY [build docker image] ******************************************************

TASK [Gathering Facts] *********************************************************
fatal: [52.5.59.113]: FAILED! => {"changed": false, "module_stderr": "/home/sjcarbon/.ssh/config line 3: Unsupported option \"rsaauthentication\"\r\nShared connection to 52.5.59.113 closed.\r\n", "module_stdout": "/bin/sh: 1: /usr/bin/python: not found\r\n", "msg": "MODULE FAILURE", "rc": 127}
```
fixed with 
```
ssh -i KEY_PATH login@location 'sudo apt-get update && sudo apt-get -y install python'
```

**Comment by kltm** [comment]
Similarly, as with AmiGO, replace the unpacking module with a manual tar `pigz` command and look to use ansible (or something, even automatic) to start the docker images up.

**Comment by kltm** [comment]
I can confirm that the "dual" CNAME feature seems to work as advertised so far!

**Comment by kltm** [comment]
Completed testing cycle: looking good!
Is it better to merge now before more work goes on here, or do you want to keep this PR open?

**Comment by abessiari** [comment]
I am working on the timeout .... Let us wait.

**Comment by abessiari** [comment]
@kltm 

done. we can discuss on the meeting. 

https://github.com/geneontology/go-graphstore/pull/19/commits/1323c82e6eb8823a628192f323223772d1d1bfc5

**Comment by kltm** [comment]
Under discussion, merging now.

**Pull request merged**